bytes/src/main/java/org/apache/tuweni/bytes/Bytes.java

```
@@ -697,26 +697,26 @@ default int toInt() {
    * @throws IllegalArgumentException if {@code size() > 4}.
    */
   default int toInt(ByteOrder order) {
-    final int numberZeros =
+    final int numberOfZeroBytes =
         order == BIG_ENDIAN ? numberOfLeadingZeroBytes() : numberOfTrailingZeroBytes();
     final int size = size();
-    final int trmSize = size - numberZeros;
-    checkArgument(trmSize <= 4, "Value of size %s has more than 4 bytes", trmSize);
-    if (trmSize == 0) {
+    if (numberOfZeroBytes == size) {
       return 0;
     }
+    final int trmSize = size - numberOfZeroBytes;
+    checkArgument(trmSize <= 4, "Value of size %s has more than 4 bytes", trmSize);
```

In fairness I think the API is a bit surprising in behavior. We fail hard if Bytes is less than 4 bytes but are tolerant if it's bigger than 4 bytes and just truncate.

IMO, I think we could just add zeros to the number if `bytes.size <= 4`